### ISSUE TITLE
ProxylessNAS: Direct Neural Architecture Search on Target Task and Hardware

### ISSUE THREAD

**Issue by sile** [sile/readings#182]
URL
-----

https://arxiv.org/abs/1812.00332

Abstract
----------

> Neural architecture search (NAS) has a great impact by automatically designing effective neural network architectures. However, the prohibitive computational demand of conventional NAS algorithms (e.g. 104 GPU hours) makes it difficult to \emph{directly} search the architectures on large-scale tasks (e.g. ImageNet). Differentiable NAS can reduce the cost of GPU hours via a continuous representation of network architecture but suffers from the high GPU memory consumption issue (grow linearly w.r.t. candidate set size). As a result, they need to utilize~\emph{proxy} tasks, such as training on a smaller dataset, or learning with only a few blocks, or training just for a few epochs. These architectures optimized on proxy tasks are not guaranteed to be optimal on the target task. In this paper, we present \emph{ProxylessNAS} that can \emph{directly} learn the architectures for large-scale target tasks and target hardware platforms. We address the high memory consumption issue of differentiable NAS and reduce the computational cost (GPU hours and GPU memory) to the same level of regular training while still allowing a large candidate set. Experiments on CIFAR-10 and ImageNet demonstrate the effectiveness of directness and specialization. On CIFAR-10, our model achieves 2.08\% test error with only 5.7M parameters, better than the previous state-of-the-art architecture AmoebaNet-B, while using 6× fewer parameters. On ImageNet, our model achieves 3.1\% better top-1 accuracy than MobileNetV2, while being 1.2× faster with measured GPU latency. We also apply ProxylessNAS to specialize neural architectures for hardware with direct hardware metrics (e.g. latency) and provide insights for efficient CNN architecture design.

Table of Contents
--------------------

- (1) Introduction
- (2) Related Work
- (3) Method
  - (3-1) Construction of Over-Parameterized Network
  - (3-2) Learning Binarized Path
    - (3-2-1) Training Binarized Architecture Parameters
  - (3-3) Handling Non-differentiable Hardware Metrics
    - (3-3-1) Making Latency Differentiable
    - (3-3-2) REINFORCE-based Approach
- (4) Experiments and Results
  - (4-1) Experiments on CIFAR-10
  - (4-2) Experiments on ImageNet
- (5) Conclusion
- (A) The List of Candidate Operations Used on CIFAR-10
- (B) Mobile Latency Prediction
- (C) Details of MnasNet's Search Cost
- (D) Implementation of the Gradient-Based Algorithm

BibTeX
------

```bibtex
@misc{cai2018proxylessnas,
    title={ProxylessNAS: Direct Neural Architecture Search on Target Task and Hardware},
    author={Han Cai and Ligeng Zhu and Song Han},
    year={2018},
    eprint={1812.00332},
    archivePrefix={arXiv},
    primaryClass={cs.LG}
}
```

See Also
----------

- GitHub: https://github.com/mit-han-lab/proxylessnas
